src/storage/event_listener.cc

```diff
@@ -59,6 +59,79 @@ std::string CompressType2String(const rocksdb::CompressionType type) {
   return "unknown";
 }
 
+std::string CompactionReason2String(const rocksdb::CompactionReason compression_reason) {
+  std::string result;
+  switch (compression_reason) {
+    case rocksdb::CompactionReason::kUnknown:
+      result = "Unknown";
+      break;
+    case rocksdb::CompactionReason::kLevelL0FilesNum:
+      result = "LevelL0FilesNum";
+      break;
+    case rocksdb::CompactionReason::kLevelMaxLevelSize:
+      result = "LevelMaxLevelSize";
+      break;
+    case rocksdb::CompactionReason::kUniversalSizeAmplification:
+      result = "UniversalSizeAmplification";
+      break;
+    case rocksdb::CompactionReason::kUniversalSizeRatio:
+      result = "UniversalSizeRatio";
+      break;
+    case rocksdb::CompactionReason::kUniversalSortedRunNum:
+      result = "UniversalSortedRunNum";
+      break;
+    case rocksdb::CompactionReason::kFIFOMaxSize:
+      result = "FIFOMaxSize";
+      break;
+    case rocksdb::CompactionReason::kFIFOReduceNumFiles:
+      result = "FIFOReduceNumFiles";
+      break;
+    case rocksdb::CompactionReason::kFIFOTtl:
+      result = "FIFOTtl";
+      break;
+    case rocksdb::CompactionReason::kManualCompaction:
+      result = "ManualCompaction";
+      break;
+    case rocksdb::CompactionReason::kFilesMarkedForCompaction:
+      result = "FilesMarkedForCompaction";
+      break;
+    case rocksdb::CompactionReason::kBottommostFiles:
+      result = "BottommostFiles";
+      break;
+    case rocksdb::CompactionReason::kTtl:
+      result = "Ttl";
+      break;
+    case rocksdb::CompactionReason::kFlush:
+      result = "Flush";
+      break;
+    case rocksdb::CompactionReason::kExternalSstIngestion:
+      result = "ExternalSstIngestion";
+      break;
+    case rocksdb::CompactionReason::kPeriodicCompaction:
+      result = "PeriodicCompaction";
+      break;
+    case rocksdb::CompactionReason::kChangeTemperature:
+      result = "ChangeTemperature";
+      break;
+    case rocksdb::CompactionReason::kForcedBlobGC:
+      result = "ForcedBlobGC";
+      break;
+    case rocksdb::CompactionReason::kRoundRobinTtl:
+      result = "RoundRobinTtl";
+      break;
+    case rocksdb::CompactionReason::kRefitLevel:
+      result = "RefitLevel";
+      break;
+    case rocksdb::CompactionReason::kNumOfReasons:
+      result = "NumOfReasons";
+      break;
+    default:
+      result = "UnSupport";
```

Comment:
I'd change this to "UnexpectedReasonValue".
In general, LGTM.
@darionyaphet thank you for the contribution.

Reply:
ok & fixed